### PR TITLE
fix fo r#869

### PR DESCRIPTION
**Pull request by rhsakarpos**
**What this PR does / why we need it**:
fix for #869 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
#869 
**Special notes for your reviewer**:
image

**Release note**:

```
This also adds a fix to delete SSE settings on bucket delete
```

### CODE REVIEW

**Comment by rhsakarpos** [comment]
@kumarashit @sfzeng @nguptaopensds @anvithks please review, thanks

File: s3/pkg/service/object.go [hunk 1]
```diff
@@ -410,6 +427,7 @@ func GetObject(ctx context.Context, req *pb.GetObjectInput, stream pb.S3_GetObje
 }
 
 func GetEncObject(ctx context.Context, req *pb.GetObjectInput, stream pb.S3_GetObjectStream, s *s3Service) error {
+
```
Comment (sfzeng) [inline s3/pkg/service/object.go]: Format code please

Comment (rhsakarpos) [reply s3/pkg/service/object.go]: did that, ran go fmt multiple times, still looks like this...am i missing something ?
image

Comment (sfzeng) [reply s3/pkg/service/object.go]: How did you run it?

Comment (rhsakarpos) [reply s3/pkg/service/object.go]: like this...

File: s3/pkg/meta/db/drivers/tidb/bucket.go [hunk 2]
```diff
@@ -478,13 +478,37 @@ func (t *TidbClient) CountObjects(ctx context.Context, bucketName, prefix string
...
+	}()
+
+	sqlTx, _ = tx.(*sql.Tx)
+
 	sqltext := "delete from buckets where bucketname=?;"
-	_, err := t.Client.Exec(sqltext, bucket.Name)
+	_, err = sqlTx.Exec(sqltext, bucket.Name)
 	if err != nil {
-		return handleDBError(err)
+		err = handleDBError(err)
```
Comment (sfzeng) [inline s3/pkg/meta/db/drivers/tidb/bucket.go]: Don't we need to return here?

Comment (rhsakarpos) [reply s3/pkg/meta/db/drivers/tidb/bucket.go]: to err on the side of caution (based on sample code i saw on a blog), did not return...yes, that might be more efficient in the sense we will not try the second SQL, but even with current code, the intent is surely met, either BOTH SQL are executed or NONE are executed. Tested this. Thanks

**Pull request merged**